test/e2e/utils.go

```diff
@@ -387,26 +387,27 @@ func getProviders() ([]TestProvider, error) {
 	return providers, nil
 }
 
-func getKubeconfig(kubeconfigPath string) (*rest.Config, error) {
-	return clientcmd.BuildConfigFromFlags("", kubeconfigPath)
-}
-
-func getKubernetesTypedClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
-	config, err := getKubeconfig(kubeconfigPath)
+func getKubernetesTypedClient() (*kubernetes.Clientset, error) {
+	restConfig, err := config.GetConfig()
 	if err != nil {
 		return nil, err
 	}
 
-	return kubernetes.NewForConfig(config)
+	return kubernetes.NewForConfig(restConfig)
 }
 
-func getKubernetesClient(kubeconfigPath string) (client.Client, error) {
-	config, err := getKubeconfig(kubeconfigPath)
+// GetKubernetesClientOrError creates and returns a kubernetes clientset or an error if creation fails
+func GetKubernetesClientOrError() (client.Client, error) {
+	var cl client.Client
+	restConfig, err := config.GetConfig()
 	if err != nil {
 		return nil, err
 	}
-
-	return client.New(config, client.Options{})
+	cl, err = client.New(restConfig, client.Options{})
+	if err != nil {
+		return nil, err
+	}
+	return cl, nil
```

Comment:
I don't see any need for making changes to this code. The previous implementation of `return client.New(config, client.Options{})` was absolutely fine and neat to read as well. The new code introduced here simply does the exact same thing that the previous code did, but with more lines of unnecessary code. Please revert this change.